### PR TITLE
correct the type of secrets

### PR DESCRIPTION
**Pull request by zhiweiyin318**
fix issue: 
https://github.com/stolostron/backlog/issues/20219
https://github.com/stolostron/backlog/issues/18580
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/controller/importconfig/cluster_info.go [hunk 1]
```diff
@@ -145,10 +145,6 @@ func getKubeAPIServerCertificate(ctx context.Context, kubeClient kubernetes.Inte
 		return nil, err
 	}
 
-	if secret.Type != corev1.SecretTypeTLS {
```
Comment (zhujian7) [inline pkg/controller/importconfig/cluster_info.go:148]: should we check all types we supported or just remove the check？

File: pkg/controller/importconfig/worker_default.go [hunk 2]
```diff
@@ -41,10 +41,25 @@ func (w *defaultWorker) generateImportSecret(ctx context.Context, managedCluster
...
 	useImagePullSecret := false
+	var imagePullSecretType corev1.SecretType
+	var imagePullSDataType string // the name without Secret is to skip gosec 101
 	imagePullSecretDataBase64 := ""
-	if imagePullSecret != nil && len(imagePullSecret.Data[".dockerconfigjson"]) != 0 {
-		imagePullSecretDataBase64 = base64.StdEncoding.EncodeToString(imagePullSecret.Data[".dockerconfigjson"])
-		useImagePullSecret = true
+	if imagePullSecret != nil {
+		switch {
+		case len(imagePullSecret.Data[".dockerconfigjson"]) != 0:
```
Comment (zhujian7) [inline pkg/controller/importconfig/worker_default.go:49]: s/`.dockerconfigjson`/`corev1.DockerConfigJsonKey`/g ？

File: pkg/controller/importconfig/worker_default.go [hunk 3]
```diff
@@ -41,10 +41,25 @@ func (w *defaultWorker) generateImportSecret(ctx context.Context, managedCluster
...
+			imagePullSDataType = ".dockerconfigjson"
+			imagePullSecretType = corev1.SecretTypeDockerConfigJson
+			imagePullSecretDataBase64 = base64.StdEncoding.EncodeToString(imagePullSecret.Data[".dockerconfigjson"])
+			useImagePullSecret = true
+		case len(imagePullSecret.Data[".dockercfg"]) != 0:
```
Comment (zhujian7) [inline pkg/controller/importconfig/worker_default.go:54]: s/`.dockercfg`/`corev1.DockerConfigKey`/g ?

**Comment by zhujian7** [comment]
If there are some ut or e2e for this, I think that will be great.

File: pkg/controller/importconfig/cluster_info.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/controller/importconfig/cluster_info.go:148]: just remove the type check, and we check tls key below

File: pkg/controller/importconfig/worker_default.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/controller/importconfig/worker_default.go]: done

File: pkg/controller/importconfig/worker_default.go (hunk 3 above)
Comment (zhiweiyin318) [reply pkg/controller/importconfig/worker_default.go]: done

**Comment by zhiweiyin318** [comment]
> If there are some ut or e2e for this, I think that will be great.

added validation in ut

**Comment by zhujian7** [comment]
> > If there are some ut or e2e for this, I think that will be great.
> 
> added validation in ut

That is great, could you add more to cover the new lines (new supported secret type opaque for cluster api certificate and new opaqua secret for image pull secret, gurnben's case)

**Pull request merged**